pkg/ccl/backupccl/testdata/backup-restore/plpgsql_procedures

```diff
@@ -0,0 +1,425 @@
+# Test backing up and restoring a database with PL/pgSQL procedures.
+new-cluster name=s
+----
+
+exec-sql
+CREATE DATABASE db1;
+USE db1;
+CREATE SCHEMA sc1;
+CREATE TABLE sc1.tbl1(a INT PRIMARY KEY);
+CREATE TYPE sc1.enum1 AS ENUM('Good');
+CREATE SEQUENCE sc1.sq1;
+CREATE PROCEDURE sc1.p1(a sc1.enum1) LANGUAGE PLpgSQL AS $$
+  BEGIN
+    SELECT a FROM sc1.tbl1;
+    SELECT nextval('sc1.sq1');
+  END
+$$;
+CREATE SCHEMA sc2;
+CREATE TABLE sc2.tbl2(a INT PRIMARY KEY);
+CREATE PROCEDURE sc2.p2(i INT) LANGUAGE PLpgSQL AS $$
+  BEGIN
+    INSERT INTO sc2.tbl2 VALUES (i);
+  END
+$$;
+----
+
+exec-sql
+CALL sc2.p2(123)
+----
+
+query-sql
+SELECT * FROM sc2.tbl2
+----
+123
+
+exec-sql
+BACKUP DATABASE db1 INTO 'nodelocal://1/test/'
+----
+
+query-sql
+WITH descs AS (
+  SHOW BACKUP LATEST IN 'nodelocal://1/test/'
+)
+SELECT database_name, parent_schema_name, object_name, object_type, is_full_cluster FROM descs
+----
+<nil> <nil> db1 database false
+db1 <nil> public schema false
+db1 <nil> sc1 schema false
+db1 sc1 tbl1 table false
+db1 sc1 enum1 type false
+db1 sc1 _enum1 type false
+db1 sc1 sq1 table false
+db1 sc1 p1 function false
+db1 <nil> sc2 schema false
+db1 sc2 tbl2 table false
+db1 sc2 p2 function false
+
+query-sql
+SELECT create_statement FROM [SHOW CREATE PROCEDURE sc1.p1]
+----
+----
+CREATE PROCEDURE sc1.p1(IN a db1.sc1.enum1)
+	LANGUAGE plpgsql
+	AS $$
+	BEGIN
+	SELECT a FROM db1.sc1.tbl1;
+	SELECT nextval('sc1.sq1'::REGCLASS);
+	END
+
+$$
+----
+----
+
+exec-sql
+CALL sc1.p1('Good'::sc1.enum1)
+----
+
+query-sql
+SELECT currval('sc1.sq1')
```

Comment:
[minor] I wonder if it would better to use nextval() somewhere before the BACKUP as another layer of validation that we are reading from teh correctly rewritten sequence desc.